sdcm/cluster.py

```diff
@@ -641,6 +642,44 @@ def _get_coredump_backtraces(self, last=True):
             self.log.error('Error retrieving core dump backtraces : %s',
                            details)
 
+    def _upload_coredump(self, coredump):
+        base_upload_url = 'scylladb-users-upload.s3.amazonaws.com/%s/%s'
+        coredump_id = os.path.basename(coredump)[:-3]
+        upload_url = base_upload_url % (coredump_id, os.path.basename(coredump))
+        self.log.info('Uploading coredump %s to %s' % (coredump, upload_url))
+        self.remoter.run("sudo curl --request PUT --upload-file "
+                         "'%s' '%s'" % (coredump, upload_url))
+        self.log.info("To download it, you may use "
+                      "'curl --user-agent [user-agent] %s > %s'",
+                      upload_url, os.path.basename(coredump))
+
+    def _get_coredump_size(self, coredump):
+        try:
+            res = self.remoter.run('stat -c %s {}'.format(coredump))
+            return int(res.stdout.strip())
+        except Exception as ex:
+            self.log.error('Failed getting coredump file size: %s', ex)
+        return None
+
+    def _split_coredump(self, coredump):
+        core_files = []
```

Comment:
The default value should be [coredump], then SCT can still try to upload the big file when it's failed to split it.